### DISCUSSION TITLE
Playbook and the '.security' field together with 'message' and 'source.ip' fields

### DISCUSSION THREAD

**Discussion by Twide** [Security-Onion-Solutions/securityonion#5088] [Unsupported Versions] [answered]
I don't know if this is a feature request, user error or an error in SO (ver 2.3.60 earlier) .

When using the playbook to build sigma detections or testing existing prebuilt rules i find that there are two fields that don't work as expected.
**message.security**
On windows logs i can't get a hit in kibana when i use the field message.security
example `message.security:logon` gives no hits, but `message:logon` gives hits.
This causes all detections on the message-field to fail when building them using the Playbook since i am forced to use the .security field. A lot of prebuilt rules also use the message field and i believe none of them would work.
Simple sigma example that translates to Message.security:
```
detection:
  keywords:
    Message:
    - '* mimikatz *'
    - '* mimilib *'
    - '* <3 eo.oe *'
```
Also, since there is a capital M in Message, it won't hit because SO uses message.security, not Message.security. Would it be possible to fix this with a small adjustment to the sigmac field mapping file?

**source.ip.security**
Almost the same thing with source.ip. All sigma rules that must use source.ip (or src_ip) as a field gets the added .security and that field does not exists with source.ip. Would the easiest fix be to also index source.ip as both IP and text, or maybe make the playbook not ad .security to fields that shouldn't have that extension?
If it was also indexed as text i believe it would be possible to use Sigmas support for "startswith"-condition (like stated here: https://github.com/Security-Onion-Solutions/securityonion/discussions/4023). I don't know if Sigma has support for ip-ranges in any other way(?).

Or am i using this wrong? (that would be the easiest fix ;)

Possible Feature request: Make the .security-field optional :) and maybe add a link to the Sigma field mapping file as a cheatsheet in Playbook as reference.

Bonus question: What is the easiest way to manually add elastalert rules so they show up nicely in the Alert-console like all other alerts? (useful when adding rules that's not supported with Sigma like a new_term-rule for example)

I have the latest version 2.3.60. Installed from ISO (at around 2.3.20 and then updated)
Everything seems to work in SO (so-status all green and so on)

Have a great weekend!

**Comment by defensivedepth** [comment, marked as answer]
RE: `.message` - this is a known issue and will be fixed soon:  https://github.com/Security-Onion-Solutions/securityonion/issues/5106

RE: IP - this was fixed in 2.3.40. Fields named src_ip / source.ip etc should be converted to source.ip.keyword instead of .security. I just verified this on a .61 install. 

There are currently no plans to make the .security subfield optional - it allows case insensitive wildcard searching which Elasticsearch does not do natively - as you run into issues, please continue to drop them here and we will get them fixed!

RE: bonus question - Drop them under `/opt/so/rules/elastalert`

**Reply by Pancakesarusrex** [reply]
when adding rules to /opt/so/elastalert  they trigger and i can find them on kibana however i do not see them on the SOC under alerts. do i have to add anything to the elastalert rule to make it populate on the SOC alerts section?

**Reply by Pancakesarusrex** [reply]
typo. I meant adding rules to  /opt/so/rules/elastalert

**Reply by Twide** [reply]
That was actually my question also, i don't think the solution is to actually just drop the rules in that directory :)
Maybe a new thread with that specific question should be created...

**Reply by defensivedepth** [reply]
Can you please post an example rule?
